src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -30,7 +30,7 @@ public static string GetEnvironmentVariableOrDefault(this IEnvironment environme
 
         public static bool IsLinuxMetricsPublishingEnabled(this IEnvironment environment)
         {
-            return environment.IsLinuxConsumption() && string.IsNullOrEmpty(environment.GetEnvironmentVariable(ContainerStartContext));
+            return environment.IsLinuxConsumptionOnAtlas() && string.IsNullOrEmpty(environment.GetEnvironmentVariable(ContainerStartContext));
```

Comment:
please see https://github.com/Azure/azure-functions-host/pull/8617#discussion_r942991081

Reply:
Since we will implement a similar metric logging mechanism for Legion, shouldn't this line also include legion runtime? So maybe something like `environment.IsLinuxConsumptionOnAtlas() || environment.IsLinuxConsumptionOnLegion()` for right now?

Reply:
yeah, I was thinking we will be modifying the existing service, so  didnt add a todo here explicitly, but we know this pending work anyway :)